execution_engine_testing/test_support/src/wasm_test_builder.rs

```diff
@@ -1168,4 +1169,27 @@ where
         self.transforms = Vec::new();
         self
     }
+
+    /// Advances eras by num_eras
+    pub fn advance_eras_by(&mut self, num_eras: u64, mut reward_items: Vec<RewardItem>) {
+        for _ in 0..num_eras {
+            let mut step_request_builder = StepRequestBuilder::new()
+                .with_parent_state_hash(self.get_post_state_hash())
+                .with_protocol_version(ProtocolVersion::V1_0_0)
+                .with_next_era_id(self.get_era().successor())
+                .with_run_auction(true);
+
+            while let Some(reward_item) = reward_items.pop() {
```

Comment:
I think this can't work. You pass in a vector of `reward_items` that you pop from, which means you can only pop so much and then you run out of elements after which you won't add any new reward items. It means that you will add reward items only for a _single era_ and not `num_eras` of iterations. And the fact that you have `while let Some(_) = reward_items.pop` hides the problem.

This means that the new code is not equivalent to the old one but also, much worse, that current tests are not good enough to find it!